src/server/main_service.cc

```diff
@@ -925,7 +955,12 @@ void Service::Init(util::AcceptServer* acceptor, std::vector<facade::Listener*>
   }
 
   // Initialize shard_set with a global callback running once in a while in the shard threads.
-  shard_set->Init(shard_num, [this] { server_family_.GetDflyCmd()->BreakStalledFlowsInShard(); });
+  shard_set->Init(shard_num, [this] {
+    server_family_.GetDflyCmd()->BreakStalledFlowsInShard();
```

Comment:
Please note that `EngineShard::RunPeriodic` runs shard_handler once in 100ms, even though the loop inside RunPeriodic is once in 1ms. 
So with this code now `server_family_.UpdateMemoryGlobalStats();` will run once in 100ms as well. I think it's not ok. I suggest that you move the 100ms check here and make it special only for `BreakStalledFlowsInShard()`  so that shard_handler will be called every iteration inside RunPeriodic.